### PR TITLE
UINV-36 Populate invoice line "Account number" and "accounting code" based on POL details	

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
The POL references an account number for the vendor that also references an accounting code. This information is critical for completing and paying invoices.
https://issues.folio.org/browse/UINV-36
## Approach
For Invoice line the POL will possibly populate account number. If not the user can select from the Organizations erpCode and/or accounts[0].appSystemNo(s). For invoice line, accounting code and account number will NOT be required and accounting code will not be editable.
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] -Code coverage on new code is 80% or greater- there is no possibility to make coverage >=80% since part of code relates to calling of ui-plugin-find-po-line and we can't test it in scope of this module.
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js [hunk 1]
```diff
@@ -78,6 +88,9 @@ class InvoiceLineEditLayer extends Component {
         }],
       };
     const hasLoaded = (!lineId || get(resources, 'invoiceLine.hasLoaded')) && get(resources, 'invoice.hasLoaded');
+    const vendor = get(resources, ['vendor', 'records', 0], {});
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js]: don't you think that you have to wait until vendor is loaded?

File: src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js [hunk 2]
```diff
@@ -94,6 +107,8 @@ class InvoiceLineEditLayer extends Component {
               parentMutator={parentMutator}
               onSubmit={this.saveInvoiceLine}
               onCancel={onCloseEdit}
+              vendorCode={vendorCode}
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js]: why it goes as a separate prop? I thought it's an attribute of invoice line

File: src/invoices/InvoicesList/Invoice/utils.js [hunk 3]
```diff
@@ -22,5 +22,6 @@ export const createInvoiceLineFromPOL = (poLine, invoiceId) => {
       quantityPhysical * get(poLine, 'cost.listUnitPrice', 0)
       + quantityElectronic * get(poLine, 'cost.quantityElectronic', 0)
     ),
+    accountingNumber: get(poLine, 'vendorDetail.vendorAccount'),
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoicesList/Invoice/utils.js]: I thought at this point you have to set account code also.

File: src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js]: fixed

File: src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js]: It's an attribute of vendor.

File: src/invoices/InvoicesList/Invoice/utils.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/invoices/InvoicesList/Invoice/utils.js]: Fixed

File: src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js [hunk 4]
```diff
@@ -77,7 +87,10 @@ class InvoiceLineEditLayer extends Component {
           percentage: 50,
         }],
       };
-    const hasLoaded = (!lineId || get(resources, 'invoiceLine.hasLoaded')) && get(resources, 'invoice.hasLoaded');
+    const hasLoaded = (!lineId || get(resources, 'invoiceLine.hasLoaded')) && get(resources, 'invoice.hasLoaded') && get(resources, 'vendor.hasLoaded');
```
Comment (NikitaSedyx) [inline src/invoices/InvoicesList/InvoiceLine/InvoiceLineEditLayer.js]: imo it should be move to function

File: src/invoices/InvoicesList/Invoice/utils.js [hunk 5]
```diff
@@ -1,11 +1,23 @@
...
-export const createInvoiceLineFromPOL = (poLine, invoiceId) => {
+export const createInvoiceLineFromPOL = (poLine, invoiceId, vendor) => {
   const quantityPhysical = get(poLine, 'cost.quantityPhysical', 0);
   const quantityElectronic = get(poLine, 'cost.quantityElectronic', 0);
+  const accountNumber = get(poLine, 'vendorDetail.vendorAccount');
+  const optionalProps = {};
+
+  if (accountNumber) {
+    const accounts = get(vendor, 'accounts', []);
+    const accountingCode = get(find(accounts, { accountNo: accountNumber }), 'appSystemNo', '') || get(vendor, 'erpCode');
```
Comment (NikitaSedyx) [inline src/invoices/InvoicesList/Invoice/utils.js]: codestyle

Comment (aliaksei-chumakou) [reply src/invoices/InvoicesList/Invoice/utils.js]: what do you mean?

Comment (NikitaSedyx) [reply src/invoices/InvoicesList/Invoice/utils.js]: long line

File: src/invoices/InvoicesList/Invoice/InvoiceDetailsLayer.js [hunk 6]
```diff
@@ -18,6 +20,14 @@ class InvoiceDetailsLayer extends Component {
       ...invoiceLinesResource,
       fetch: false,
     },
+    vendor: {
```
Comment (NikitaSedyx) [inline src/invoices/InvoicesList/Invoice/InvoiceDetailsLayer.js]: it's strange, but previously I commented this as `copy paste`, but comment disappeared

File: src/invoices/InvoicesList/Invoice/utils.js (hunk 5 above)
Comment (NikitaSedyx) [reply src/invoices/InvoicesList/Invoice/utils.js]: line is changed - comment doesn't make sense

**Pull request merged**